daemon-scheduler/pkg/regex/regex_consts.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package regex
+
+const (
+	clusterNameRegexWithoutStart = "[a-zA-Z][a-zA-Z0-9_-]{1,254}$"
+	ClusterNameRegex             = "^" + clusterNameRegexWithoutStart
+	ClusterARNRegex              = "^(arn:aws:ecs:)([\\-\\w]+):[0-9]{12}:(cluster)/" + clusterNameRegexWithoutStart
```

Comment:
Why are there capture groups in this regexp that are never used?  Is and will account-id always be exactly 12 digits? Can regions really start with `-`? If this ties to a published spec [eg: aws-arns-and-namespaces](http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arn-syntax-ecs), it would be good to link it in a comment.